staging/src/k8s.io/dynamic-resource-allocation/resourceslice/resourceslicecontroller_test.go

```diff
@@ -0,0 +1,276 @@
+	}
+
+	// Create the Controller instance
+	ctrl := &Controller{
+		cancel:     cancel,
+		kubeClient: kubeClient,
+		driver:     testDriveName,
+		owner:      owner,
+		queue: workqueue.NewTypedRateLimitingQueueWithConfig(
+			workqueue.DefaultTypedControllerRateLimiter[string](),
+			workqueue.TypedRateLimitingQueueConfig[string]{Name: "node_resource_slices"},
+		),
+		mutex: sync.RWMutex{},
+	}
+
+	// Set up informerFactory and sliceStore
+	informerFactory := informers.NewSharedInformerFactoryWithOptions(kubeClient, 0,
+		informers.WithTweakListOptions(func(options *metav1.ListOptions) {
+			// Set filtering criteria
+			selector := fields.Set{resourceapi.ResourceSliceSelectorDriver: ctrl.driver}
+			if ctrl.owner.APIVersion == "v1" && ctrl.owner.Kind == "Node" {
+				selector[resourceapi.ResourceSliceSelectorNodeName] = ctrl.owner.Name
+			}
+			options.FieldSelector = selector.String()
+		}))
+
+	resourceSliceInformer := informerFactory.Resource().V1alpha3().ResourceSlices().Informer()
+	err := resourceSliceInformer.AddIndexers(cache.Indexers{
+		poolNameIndex: func(obj interface{}) ([]string, error) {
+			slice, ok := obj.(*resourceapi.ResourceSlice)
+			if !ok {
+				return []string{}, nil
+			}
+			return []string{slice.Spec.Pool.Name}, nil
+		},
+	})
+	if err != nil {
+		return nil, nil
+	}
+
+	ctrl.resources = inputDriverResources
+	ctrl.sliceStore = resourceSliceInformer.GetIndexer()
+	return ctrl, informerFactory
+}
+
+func newResourceSlice(name, nodeName, driveName, poolName string) *resourceapi.ResourceSlice {
+	return &resourceapi.ResourceSlice{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+			UID:  types.UID(name),
+		},
+		Spec: resourceapi.ResourceSliceSpec{
+			NodeName: nodeName,
+			Driver:   driveName,
+			Pool: resourceapi.ResourcePool{
+				Name:               poolName,
+				ResourceSliceCount: 1,
+			},
+		},
+	}
+}
+
+func newExpectResourceSlice(name, nodeName, nodeUID, driveName, poolName, deviceName string, generateName bool) *resourceapi.ResourceSlice {
+	resourceSlice := &resourceapi.ResourceSlice{
+		ObjectMeta: metav1.ObjectMeta{
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion: "v1",
+					Kind:       "Node",
+					Name:       nodeName,
+					UID:        types.UID(nodeUID),
+					Controller: ptr.To(true),
+				},
+			},
+		},
+		Spec: resourceapi.ResourceSliceSpec{
+			NodeName: nodeName,
+			Driver:   driveName,
+			Pool: resourceapi.ResourcePool{
+				Name:               poolName,
+				ResourceSliceCount: 1,
+			},
+			Devices: []resourceapi.Device{{Name: deviceName}},
+		},
+	}
+
+	if generateName {
+		resourceSlice.ObjectMeta.GenerateName = name
+	} else {
+		resourceSlice.ObjectMeta.Name = name
+		resourceSlice.ObjectMeta.UID = types.UID(name)
+		resourceSlice.ObjectMeta.OwnerReferences = []metav1.OwnerReference{
+			{
+				APIVersion: "v1",
+				Kind:       "Node",
+				Name:       nodeName,
+				UID:        types.UID(nodeUID),
+				Controller: ptr.To(true),
+			},
+		}
```

Comment:
The `OwnerReference` is already set in lines 238-246, so I don't think we need to set it again.

```suggestion

```

Reply:
thanks! sgtm